src/test/java/com/stripe/BaseStripeTest.java

```diff
@@ -1,170 +1,481 @@
+   *     APIResource.RequestType, RequestOptions, Class, String)
+   */
+  public static <T> void stubRequest(
+      APIResource.RequestMethod method,
+      String path,
       Class<T> clazz,
-      String url,
-      Map<String, Object> params,
-      RequestOptions requestOptions) throws StripeException {
-    verifyRequest(APIResource.RequestMethod.DELETE, clazz, url, params,
-        APIResource.RequestType.NORMAL, requestOptions);
+      String response) throws StripeException {
+    stubRequest(method, path, null, null, null, clazz, response);
   }
 
-  public static <T> void verifyDelete(
+  /**
+   * {@code requestType} and {@code options} defaults to {@code null}.
+   *
+   * @see BaseStripeTest#stubRequest(APIResource.RequestMethod, String, Map,
+   *     APIResource.RequestType, RequestOptions, Class, String)
+   */
+  public static <T> void stubRequest(
+      APIResource.RequestMethod method,
+      String path,
+      Map<String, Object> params,
       Class<T> clazz,
-      String url) throws StripeException {
-    verifyRequest(APIResource.RequestMethod.DELETE, clazz, url, null,
-        APIResource.RequestType.NORMAL, RequestOptions.getDefault());
+      String response) throws StripeException {
+    stubRequest(method, path, params, null, null, clazz, response);
   }
 
   /**
-   * Verifies that the specified request occurred.
+   * Stubs an API request. This should rarely be necessary, but some endpoints are not
+   * supported by stripe-mock yet.
+   *
+   * @param method HTTP method (GET, POST or DELETE)
+   * @param path request path (e.g. "/v1/charges"). Can also be an abolute URL.
+   * @param params map containing the parameters. If null, the parameters are not checked.
+   * @param requestType request type (NORMAL or MULTIPART). If null, the request type is not
+   *     checked.
+   * @param options request options. If null, the options are not checked.
+   * @param clazz Class of the API resource that will be returned for the stubbed request.
+   * @param response JSON payload of the API resource that will be returned for the stubbed
+   *     request.
    */
-  public static <T> void verifyRequest(
+  public static <T> void stubRequest(
       APIResource.RequestMethod method,
-      Class<T> clazz,
-      String url,
+      String path,
       Map<String, Object> params,
       APIResource.RequestType requestType,
-      RequestOptions options) throws StripeException {
-    verify(networkMock).request(
-        eq(method),
-        eq(url),
-        argThat(new ParamMapMatcher(params)),
-        eq(clazz),
-        eq(requestType),
-        argThat(new RequestOptionsMatcher(options)));
+      RequestOptions options,
+      Class<T> clazz,
+      String response) throws StripeException {
+    String url;
+    if (path.startsWith("/")) {
+      url = String.format("%s%s", Stripe.getApiBase(), path);
+    } else {
+      url = path;
+    }
+
+    Mockito
+        .doReturn(APIResource.GSON.fromJson(response, clazz))
```

Comment:
Is this indentation right? Seems weird that the style is two spaces everywhere else, but then four spaces when chaining method calls like this.